### PR TITLE
[CALCITE-3176] File adapter for parsing JSON files

### PR DESCRIPTION
**Pull request by XuQianJin-Stars**
Fix ISSUE [CALCITE-3176](https://issues.apache.org/jira/browse/CALCITE-3176)

### CODE REVIEW

**Comment by XuQianJin-Stars** [comment]
I am still optimizing this PR now.

**Comment by michaelmior** [comment]
@XuQianJin-Stars Could you rebase into a single commit? This makes things much easier to review.

**Comment by XuQianJin-Stars** [comment]
hi @michaelmior Thank you very much, Comments addressed.

File: example/csv/src/main/java/org/apache/calcite/adapter/csv/JsonTable.java [hunk 1]
```diff
@@ -16,46 +16,91 @@
...
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
 
-  public String toString() {
-    return "JsonTable";
+    if (jsonObj == null) {
+      list = new ArrayList<>();
+      jsonFieldMap.put("EmptyFileHasNoColumns", Boolean.TRUE);
```
Comment (michaelmior) [inline example/csv/src/main/java/org/apache/calcite/adapter/csv/JsonTable.java]: I haven't tested this, but is it not possible to actually have a table with no columns?

Comment (XuQianJin-Stars) [reply example/csv/src/main/java/org/apache/calcite/adapter/csv/JsonTable.java]: hi @michaelmior I am referring to the logic implementation of this CSV.  [CsvEnumerator.java#L121](https://github.com/apache/calcite/blob/58b6ad6135d909c053f15fb902d8b73559d1fa53/example/csv/src/main/java/org/apache/calcite/adapter/csv/CsvEnumerator.java#L121)

Comment (michaelmior) [reply example/csv/src/main/java/org/apache/calcite/adapter/csv/JsonTable.java]: Thanks for pointing that out. I didn't realize that logic was already in there. This looks good to me!

**Pull request merged**